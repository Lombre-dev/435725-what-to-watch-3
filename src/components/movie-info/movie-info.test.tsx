import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MovieInfo from './movie-info';

const MOVIE = {
  id: 0,
  title: `The Grand Budapest Hotel`,
  genres: [`Drama`],
  year: 2014,
  poster: `/img/the-grand-budapest-hotel-poster.jpg`,
  frames: [`/img/the-grand-budapest-hotel-poster.jpg`],
  ratingScore: 8.9,
  ratingReviewsCount: 240,
  preview: `/samples/sintel_trailer-480p.mp4`,
  src: `/samples/sintel_trailer-480p.mp4`,
  description: `Description`,
  story: `Story`,
  director: `Wes Andreson`,
  duration: 99,
  actors: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Some Actor 1`,
    `Some Actor 2`,
  ],
};

const ACTIVE_TAB = 0;
const HANDLE_EVENT = () => {
  // eslint-disable-next-line no-console
  console.log(`handleTestEvent`);
};

describe(`<MovieDetailInfo />`, () => {

  it(`render should be match markup`, () => {

    const result = renderer
      .create(<MovieInfo
        movie={MOVIE}
        activeTab={ACTIVE_TAB}
        onTabClick={HANDLE_EVENT}
      />)
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
