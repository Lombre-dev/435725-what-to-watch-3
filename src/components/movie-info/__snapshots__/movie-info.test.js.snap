// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetailInfo /> render should be match markup 1`] = `
<div
  className="movie-card__wrap movie-card__translate-top"
>
  <div
    className="movie-card__info"
  >
    <div
      className="movie-card__poster movie-card__poster--big"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="/img/the-grand-budapest-hotel-poster.jpg"
        width="218"
      />
    </div>
    <div
      className="movie-card__desc"
    >
      <nav
        className="movie-nav movie-card__nav"
      >
        <ul
          className="movie-nav__list"
        >
          <li
            className="movie-nav__item movie-nav__item--active"
          >
            <a
              className="movie-nav__link"
              data-index={0}
              href="#"
              onClick={[Function]}
            >
              Overview
            </a>
          </li>
          <li
            className="movie-nav__item "
          >
            <a
              className="movie-nav__link"
              data-index={1}
              href="#"
              onClick={[Function]}
            >
              Details
            </a>
          </li>
          <li
            className="movie-nav__item "
          >
            <a
              className="movie-nav__link"
              data-index={2}
              href="#"
              onClick={[Function]}
            >
              Reviews
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="movie-rating"
      >
        <div
          className="movie-rating__score"
        >
          8,9
        </div>
        <p
          className="movie-rating__meta"
        >
          <span
            className="movie-rating__level"
          >
            Very good
          </span>
          <span
            className="movie-rating__count"
          >
            240 ratings
          </span>
        </p>
      </div>
      <div
        className="movie-card__text"
      >
        <p>
          Description
        </p>
        <p>
          Story
        </p>
        <p
          className="movie-card__director"
        >
          <strong>
            Director: 
            Wes Andreson
          </strong>
        </p>
        <p
          className="movie-card__starring"
        >
          <strong>
            Starring: 
            Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other
          </strong>
        </p>
      </div>
    </div>
  </div>
</div>
`;
